web_m2x_options/static/src/components/form.esm.js

```diff
@@ -0,0 +1,560 @@
+/** @odoo-module **/
+
+import {
+    Many2ManyTagsField,
+    Many2ManyTagsFieldColorEditable,
+    many2ManyTagsField,
+} from "@web/views/fields/many2many_tags/many2many_tags_field";
+import {Many2OneField, many2OneField} from "@web/views/fields/many2one/many2one_field";
+import {X2ManyField, x2ManyField} from "@web/views/fields/x2many/x2many_field";
+
+import {Dialog} from "@web/core/dialog/dialog";
+import {FormController} from "@web/views/form/form_controller";
+import {FormViewDialog} from "@web/views/view_dialogs/form_view_dialog";
+import {Many2OneAvatarField} from "@web/views/fields/many2one_avatar/many2one_avatar_field";
+import {Many2OneBarcodeField} from "@web/views/fields/many2one_barcode/many2one_barcode_field";
+import {Many2XAutocomplete} from "@web/views/fields/relational_utils";
+import {ReferenceField} from "@web/views/fields/reference/reference_field";
+import {_t} from "@web/core/l10n/translation";
+import {evaluateBooleanExpr} from "@web/core/py_js/py";
+import {isX2Many} from "@web/views/utils";
+import {is_option_set} from "@web_m2x_options/components/relational_utils.esm";
+import {patch} from "@web/core/utils/patch";
+import {session} from "@web/session";
+import {sprintf} from "@web/core/utils/strings";
+import {useService} from "@web/core/utils/hooks";
+
+const {Component} = owl;
+
+/**
+ *  Patch Many2ManyTagsField
+ **/
+patch(Many2ManyTagsField.prototype, {
+    setup() {
+        super.setup();
+        this.actionService = useService("action");
+    },
+    /**
+     * @override
+     */
+    getTagProps(record) {
+        const props = super.getTagProps(record);
+        props.onClick = (ev) => this.onMany2ManyBadgeClick(ev, record);
+        return props;
+    },
+    async onMany2ManyBadgeClick(event, record) {
+        var self = this;
+        if (self.props.open) {
+            var context = self.context;
+            var id = record.data.id;
+            if (self.props.readonly) {
+                event.preventDefault();
+                event.stopPropagation();
+                const action = await self.orm.call(
+                    self.props.relation,
+                    "get_formview_action",
+                    [[id]],
+                    {context: context}
+                );
+                self.actionService.doAction(action);
+            } else {
+                const view_id = await self.orm.call(
+                    self.props.relation,
+                    "get_formview_id",
+                    [[id]],
+                    {context: context}
+                );
+
+                const write_access = await self.orm.call(
+                    self.props.relation,
+                    "check_access_rights",
+                    [],
+                    {operation: "write", raise_exception: false}
+                );
+                var can_write = self.props.canWrite;
+                self.dialog.add(FormViewDialog, {
+                    resModel: self.props.relation,
```

Comment:
I think it should be replaced by resModel: record.resModel

Reply:
Correct, the function was broken. It should be fixed now.